### PR TITLE
feat(wizard): add Telegram onboarding to the setup wizard

### LINKED ISSUE #1481: Onboarding: add Telegram to the wizard (currently no _configure_telegram)
## Problem

`app/cli/wizard/flow.py` ships `_configure_slack` and `_configure_discord`, but **there is no `_configure_telegram`** — even though the rest of the stack already supports Telegram:

- Verifier: `app/integrations/_verification_adapters.py::_verify_telegram` (calls `https://api.telegram.org/bot<token>/getMe` and reports `Connected to Telegram bot @<username>`).
- Config model: `TelegramBotConfig` in `app/integrations/config_models.py` (`bot_token`, optional `default_chat_id`).
- Delivery helper: `app/utils/telegram_delivery.py::send_telegram_report` (reads `bot_token` / `chat_id` from a `telegram_ctx`).

As a result, operators have to discover `TELEGRAM_BOT_TOKEN` / `TELEGRAM_DEFAULT_CHAT_ID` by reading the source. They never see Telegram in the integration picker (`_configure_selected_integrations`, `flow.py:1457+`), the onboarding wizard never validates the token, the credential is never persisted via `upsert_integration("telegram", ...)`, and `.env` is never synced.

## Proposed change

1. Add `_configure_telegram()` to `app/cli/wizard/flow.py`, modeled on `_configure_discord` (lines 1333–1385):
   - Prompt for `bot_token` (secret) and an optional `default_chat_id`.
   - Call the existing `_verify_telegram` verifier (or expose a thin `validate_telegram_integration` wrapper next to the other `validate_*_integration` helpers in the wizard validators) and render the result via `_render_integration_result("Telegram", result)`.
   - On success, `upsert_integration("telegram", {"credentials": {"bot_token": ..., "default_chat_id": ...}})` and `sync_env_values({"TELEGRAM_BOT_TOKEN": ..., "TELEGRAM_DEFAULT_CHAT_ID": ...})`.
2. Add a `Choice(value="telegram", label="Telegram", hint="Post findings to a Telegram chat")` to the `integration_choices` list in `_configure_selected_integrations` and wire it into the `handlers` dict alongside `"discord": _configure_discord`.
3. Surface Telegram in any "currently configured integrations" status output that already lists Slack/Discord.
4. Tests:
   - `tests/cli/wizard/` — verify the new branch prompts for token + chat_id, calls the verifier, persists credentials via `upsert_integration`, and writes the two env vars.
   - `tests/integrations/test_verify.py` — confirm the wizard path exercises `_verify_telegram` (mock the `getMe` HTTP call).
5. Docs: add a `docs/telegram.mdx` (or extend an existing messaging integrations doc) covering BotFather setup, `chat_id` discovery, and the env vars produced by the wizard.

## Acceptance criteria

- Running `opensre onboard` (or the integration submenu) shows **Telegram** in the picker.
- Selecting it prompts for token + chat id, validates with `_verify_telegram`, persists via `upsert_integration("telegram", ...)`, and writes `TELEGRAM_BOT_TOKEN` / `TELEGRAM_DEFAULT_CHAT_ID` to `.env`.
- `make lint`, `make typecheck`, `make test-cov` all pass.

## Related code

- `app/cli/wizard/flow.py` (`_configure_slack` ~line 657, `_configure_discord` ~line 1333, `_configure_selected_integrations` ~line 1450)
- `app/integrations/_verification_adapters.py` (`_verify_telegram` line 310)
- `app/integrations/config_models.py` (`TelegramBotConfig` line 644)
- `app/utils/telegram_delivery.py`

### PR DESCRIPTION
**Pull request by cloudenochcsis**
Fixes #1481

#### Describe the changes you have made in this PR -

- added a Telegram onboarding path to `opensre onboard`, including token validation, integration persistence, and `.env` sync
- exposed `validate_telegram_integration` through the wizard validator surface by reusing the existing Telegram verifier
- redacted Telegram bot tokens from onboarding validation failures so bad probes do not leak credentials to the terminal
- added wizard, verifier, docs-nav, and interactive smoke coverage for the new Telegram option
- added `docs/telegram.mdx` and wired it into `docs/docs.json`

### Demo/Screenshot for feature changes and bug fixes - 

```text
$ make lint && make format-check && make typecheck && make test-cov
...
5416 passed, 5 skipped, 133 warnings in 70.78s
```

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

I kept the change narrow to the onboarding gap already described in #1481. OpenSRE already had Telegram delivery, catalog/env resolution, and the shared `_verify_telegram` implementation, so I reused that verifier through a small wizard-facing validator instead of introducing a second Telegram-specific probe path. The wizard now prompts for the bot token and optional default chat ID, validates the token before saving, persists the Telegram credentials, and syncs the matching environment variables. I also updated the interactive smoke tests for the extra menu option, added direct regression coverage for the new validator path, and wired the new Telegram doc into the docs navigation so the setup guide is discoverable.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

### CODE REVIEW

File: app/cli/wizard/integration_validators/http_probe_validators.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import httpx
 
 from app.integrations.models import SlackWebhookConfig
+from app.integrations.verify import _verify_telegram
```
[the remark this answers is not in the document]
Comment (cloudenochcsis) [reply app/cli/wizard/integration_validators/http_probe_validators.py]: Fixed in e39dd249: `validate_telegram_integration()` now makes the `getMe` request directly with `httpx`, keeps the token redaction local to the onboarding validator, and no longer imports `_verify_telegram`.

File: docs/docs.json [hunk 2]
```diff
@@ -110,15 +110,16 @@
...
+                "discord",
+                "github",
+                "gitlab",
+                "google-docs",
+                "jira",
+                "slack",
+                "telegram",
+                "vercel"
+              ]
+            },
```
[the remark this answers is not in the document]
Comment (cloudenochcsis) [reply docs/docs.json]: Fixed in e39dd249: the `docs/docs.json` integration nav block is re-indented consistently and the file still parses cleanly.

**Comment by cloudenochcsis** [comment]
@greptile review

**Comment by muddlebee** [comment]
@cloudenochcsis demo is missing and resolve conflicts. pls

**Review by yashksaini-coder — CHANGES_REQUESTED** [review]
Reviewed the wizard flow + validator code carefully. The wizard path itself (`http_probe_validators.py`) correctly redacts the bot token with `str(err).replace(telegram_config.bot_token, "<redacted>")`. Test coverage for the wizard happy path + redaction is in place. **One real issue blocks merge.**

## Critical: Token leak in the CLI verify path

`app/integrations/_verification_adapters.py` `_verify_telegram` (around line 320):

```python
except Exception as exc:
    return result("telegram", source, "failed", f"Telegram API check failed: {exc}")
```

The URL passed to `requests.get` is `https://api.telegram.org/bot{bot_token}/getMe` — the bot token is embedded in the URL path. Any network-level exception from `requests` (connection error, redirect error, SSL error, timeout) renders `str(exc)` containing the URL verbatim, leaking the token into the verifier output displayed by `opensre integrations verify telegram`.

The wizard validator at `http_probe_validators.py:155` already shows the right pattern:

```python
detail = str(err).replace(telegram_config.bot_token, "<redacted>")
return IntegrationHealthResult(ok=False, detail=f"Telegram API unreachable: {detail}")
```

Apply the same redaction in `_verify_telegram`:

```python
except Exception as exc:
    detail = str(exc).replace(bot_token, "<redacted>")
    return result("telegram", source, "failed", f"Telegram API check failed: {detail}")
```

## Test gap

`tests/integrations/test_verify.py` covers only the happy path for `_verify_telegram`. A failure-path test mirroring `test_validate_telegram_integration_redacts_bot_token_on_error` from the wizard tests would have caught this and prevents future regressions. One test, easy add.

## Pluses

- Wizard flow `_configure_telegram()` cleanly mirrors `_configure_discord()`.
- Empty `default_chat_id` -> empty string in `.env` -> `or None` coerce in `_catalog_impl.py` is consistent and correct.
- `getMe`-based probe with bot-username surfacing is the right choice for a verification UX.
- Token redaction in wizard validator + tests covering it shows the team thought about this — the gap is just that the same redaction wasn't applied to the second code path handling the same secret.

## Cross-checks performed

- Read `_verify_telegram` body and confirmed no redaction.
- Read `http_probe_validators.py` wizard path and confirmed the redaction template.
- Verified the URL embeds the token (`https://api.telegram.org/bot{bot_token}/getMe`).
- Confirmed test coverage exists for the wizard path but not the CLI verify path.

## Verdict

Request changes — one-line redaction fix in `_verify_telegram` plus a failure-path test, then this is a clean APPROVE.
